### ISSUE TITLE
Saved views not loading

### ISSUE THREAD

**Issue by rsmuc** [astubenbord/paperless-mobile#69]
![grafik](https://user-images.githubusercontent.com/6809635/212278624-ee1b2b71-1a84-4043-b796-01297fa6e6c9.png)

Saved views are not loading. See screeshot.

Paperless Server-Version: 1.9.2

Paperless Mobile: 1.5.0+13

**Comment by astubenbord** [comment]
Hi, does this always occur or just sometimes?

**Comment by rsmuc** [comment]
> Hi, does this always occur or just sometimes?

Always. Never did see it working.

**Comment by astubenbord** [comment]
Okay, I'll see if it has something to do with the paperless version, maybe some parsing issue

**Comment by astubenbord** [comment]
Doesn't seem to be a version issue, but it's hard to debug without knowing what rules you have stored in your views. Maybe some characters the app cannot parse, but not sure...

**Comment by snadorp** [comment]
I'm seeing the same issue here with a fresh paperless-ngx install. So still very small scoped. Any info i can supply?

**Comment by astubenbord** [comment]
You could try to attach a debugger using `adb logcat *:S flutter:V` and see if there is a call to /api/saved_views/, otherwise I suppose it's hard for me to debug.

**Comment by snadorp** [comment]
I'm currently running `Paperless-ngx v. 1.11.3` 

A `GET`  on `/paperless/api/saved_views/` returns
```json
{
    "count": 1,
    "next": null,
    "previous": null,
    "results": [
        {
            "id": 1,
            "name": "Inbox",
            "show_on_dashboard": true,
            "show_in_sidebar": false,
            "sort_field": null,
            "sort_reverse": false,
            "filter_rules": [
                {
                    "rule_type": 5,
                    "value": null
                }
            ]
        }
    ]
}
```

The logs return:

```
01-21 20:38:36.486 17411 17432 I flutter : ╔╣ Response ║ GET ║ Status: 200 OK
01-21 20:38:36.486 17411 17432 I flutter : ║  https://THEHOST/api/saved_views/
01-21 20:38:36.486 17411 17432 I flutter : ╚══════════════════════════════════════════════════════════════════════════════════════════╝
01-21 20:38:36.487 17411 17432 E flutter : [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: Bad state: No element
01-21 20:38:36.487 17411 17432 E flutter : #0      Iterable.first (dart:core/iterable.dart:624)
01-21 20:38:36.487 17411 17432 E flutter : #1      new SavedView.fromJson (package:paperless_api/src/models/saved_view_model.dart:52)
01-21 20:38:36.487 17411 17432 E flutter : #2      _collectionFromJson.<anonymous closure> (package:paperless_api/src/request_utils.dart:75)
01-21 20:38:36.487 17411 17432 E flutter : #3      MappedIterable.elementAt (dart:_internal/iterable.dart:375)
01-21 20:38:36.487 17411 17432 E flutter : #4      ListIterator.moveNext (dart:_internal/iterable.dart:344)
01-21 20:38:36.487 17411 17432 E flutter : #5      new _GrowableList._ofEfficientLengthIterable (dart:core-patch/growable_array.dart:189)
01-21 20:38:36.487 17411 17432 E flutter : #6      new _GrowableList.of (dart:core-patch/growable_array.dart:150)
01-21 20:38:36.487 17411 17432 E flutter : #7      new List.of (dart:core-patch/array_patch.dart:47)
01-21 20:38:36.487 17411 17432 E flutter : #8      SetMixin.toList (dart:collection/set.dart:118)
01-21 20:38:36.487 17411 17432 E flutter : #9      _collectionFromJson (package:paperless_api/src/request_utils.dart:75)
01-21 20:38:36.487 17411 17432 E flutter : #10     compute.<anonymous closure> (package:flutter/src/foundation/_isolates_io.dart:19)
01-21 20:38:36.487 17411 17432 E flutter : #11     _RemoteRunner._run (dart:isolate:954)
01-21 20:38:36.487 17411 17432 E flutter : #12     _delayEntrypointInvocation.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:300)
01-21 20:38:36.487 17411 17432 E flutter : #13     _RawReceivePort._handleMessage (dart:isolate-patch/isolate_patch.dart:192)
01-21 20:38:36.487 17411 17432 E flutter :
```

**Comment by astubenbord** [comment]
Thanks! I think this will help, I'll look into it

**Comment by astubenbord** [comment]
Okay, I see why this happens. Could you select this view in the paperless web client and tell me by which field your documents are sorted? I assumed the sort field is always non-null

**Comment by snadorp** [comment]
<img width="1244" alt="Screenshot 2023-01-21 at 20 50 49" src="https://user-images.githubusercontent.com/609348/213884696-44d23a1b-2551-44f6-a459-16b9d6498925.png">

**Comment by snadorp** [comment]
I guess that's the problem then, the model defines it as nullable: https://github.com/paperless-ngx/paperless-ngx/blob/main/src/documents/models.py#L382

**Comment by astubenbord** [comment]
Ah I see, did you create this via the django interface? That's interesting, I think you cannot create a view without an explicit sort field in the paperless web client. Even in the official UI source code it is defined as non-nullable: https://github.com/paperless-ngx/paperless-ngx/blob/d94ca2962ebe2758b80d9deae01a20953f0edc46/src-ui/src/app/data/paperless-saved-view.ts#L4. I just checked and when loading the saved view, the sort field remains unset even in the webclient.

**Comment by snadorp** [comment]
Yes indeed, I created it in Django Admin, as I did not see another way when I installed Paperless.
To test this I just created another view in via the web interface. It looks like this:

```json
{
            "id": 2,
            "name": "Inbox items",
            "show_on_dashboard": true,
            "show_in_sidebar": true,
            "sort_field": null,
            "sort_reverse": false,
            "filter_rules": [
                {
                    "rule_type": 6,
                    "value": "2"
                }
            ]
```

Sortfield is also `null`. I just selected a tag and saved the view.

**Comment by astubenbord** [comment]
You're right, sorry, I can reproduce this now, I'll fix this and publish an update :)

**Comment by snadorp** [comment]
Awesome thank you!

**Comment by astubenbord** [comment]
Resolved in [v1.5.3](https://github.com/astubenbord/paperless-mobile/releases/tag/v1.5.3-beta)
